app/models/user_mute.rb

```diff
@@ -0,0 +1,45 @@
+# == Schema Information
+#
+# Table name: user_mutes
+#
+#  id           :bigint(8)        not null, primary key
+#  creator_id   :bigint(8)        not null
+#  appointee_id :bigint(8)        not null
+#  created_at   :datetime         not null
+#  updated_at   :datetime         not null
+#
+# Indexes
+#
+#  index_user_mutes_on_appointee_id                 (appointee_id)
+#  index_user_mutes_on_creator_id                   (creator_id)
+#  index_user_mutes_on_creator_id_and_appointee_id  (creator_id,appointee_id) UNIQUE
+#
+# Foreign Keys
+#
+#  fk_rails_...  (appointee_id => users.id)
+#  fk_rails_...  (creator_id => users.id)
+#
+class UserMute < ApplicationRecord
+  belongs_to :creator, :class_name => "User"
+  belongs_to :appointee, :class_name => "User"
+
+  validates :appointee, :uniqueness => { :scope => :creator_id }
+
+  def self.for_message?(message)
+    sender = message.sender
+
+    !sender.administrator? &&
+      !sender.moderator? &&
+      active_for?(
+        :current_user => message.recipient,
+        :other_user => sender
+      )
+  end
+
+  def self.active_for?(current_user:, other_user:)
```

Comment:
Why use different names for the users in the argument list here? It's the same concept as the member names so why not use the same names?

Of course once you do that the question becomes is this method even needed when it's just forwarding the arguments to `exists?`?

Reply:
> Of course once you do that the question becomes is this method even needed when it's just forwarding the arguments to exists??

good point. Initially there were more places from which `UserMute.active_for?` in which the the terms `current_user` and `other_user` are used to reference the instances. So I thought it was making the code a bit easier to read/grasp from that context.

Now there is just a single call, plus the call from `UserMute#for_message?` – I think it's ok to drop `UserMute#active_for?` and just call `UserMute#exists?` directly 👌🏽